sklearn/decomposition/pca.py

```diff
@@ -417,6 +417,12 @@ def _fit_full(self, X, n_components):
                              "min(n_samples, n_features)=%r with "
                              "svd_solver='full'"
                              % (n_components, min(n_samples, n_features)))
+        elif n_components >= 1:
+            if not np.issubdtype(type(n_components), np.integer):
```

Comment:
Completely nitpicking (and I might even be wrong), but I think that it reads better like this:
```py
if not isinstance(n_components, (int, np.integer)):
```

Reply:
Ye, just double checked and that is the same, and seems more elegant. I'll change it.